sql/2020/03_Markup/pages_almanac_by_device_and_attribute_name_present.sql

```diff
@@ -0,0 +1,46 @@
+#standardSQL
+# pages almanac metrics grouped by device and element attributes being used (present)
+
+CREATE TEMP FUNCTION AS_PERCENT (freq FLOAT64, total FLOAT64) RETURNS FLOAT64 AS (
+  ROUND(SAFE_DIVIDE(freq, total), 4)
+);
+
+CREATE TEMPORARY FUNCTION get_almanac_attribute_names(almanac_string STRING)
+RETURNS ARRAY<STRING> LANGUAGE js AS '''
+try {
+    var almanac = JSON.parse(almanac_string); 
+
+    if (Array.isArray(almanac) || typeof almanac != 'object') return [];
+
+    if (almanac.attributes_used_on_elements) {
+      return Object.keys(almanac.attributes_used_on_elements); 
+    }
+    
+} catch (e) {
+    
+}
+return []; 
+''';
+
+SELECT
+  _TABLE_SUFFIX AS client,
+  attribute_name,
```

Comment:
Limit the results, eg `LIMIT 1000` or `HAVING pages >= 100`. I think there would be a lot of unusable data in the tail < 100 pages.

Reply:
Changed to 1,000 and added the LIMIT to the second one.